sql/parser/datum.go

```diff
@@ -337,6 +337,12 @@ type DDate struct {
 	time.Time
 }
 
+// MakeDDate constructs a DDate from a time.Time.
+func MakeDDate(t time.Time) DDate {
+	year, month, day := t.Date()
+	return DDate{Time: time.Date(year, month, day, 0, 0, 0, 0, t.Location())}
```

Comment:
Btw, we're using `time.Truncate` in the implementation of `current_date`. Does that need to be fixed?